### ISSUE TITLE
Base.many has some problem with repeated characters

### ISSUE THREAD

**Issue by tprimke** [bitwalker/combine#3]
While working with the Combine parser, I've found the strange behaviour presented below:

```
import Combine.Parsers.Base; import Combine.Parsers.Text
p = many char
Combine.parse "abcde", p
> [["a", "b", "c", "d", "e"]]
Combine.parse "aaaaa", p
> [["a", "a", "a"]]

p = many char("a")
Combine.parse "abcde", p
> [["a"]]
Combine.parse "aaabb", p
> [["a", "a"]]
Combine.parse "aaaaa", p
> [["a", "a", "a"]]
```

Is this correct? (If so, why? I expected all the "a"s to be parsed...)

I use Elixir 1.1.1 (Erlang/OTP 18), and Combine 0.6.0.

**Comment by bitwalker** [comment]
I believe I know why this is happening, I'll look into it asap. In the meantime, depending on your needs, you could use `word_of(~r/[a]/)` which will return all of the a's as a single string. I'll post back here when I have a fix pushed.

**Comment by bitwalker** [comment]
Fixed in master and in 0.7.0. Thanks for reporting!
